staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -518,6 +518,14 @@ func InClusterConfig() (*Config, error) {
 	if len(host) == 0 || len(port) == 0 {
 		return nil, ErrNotInCluster
 	}
+	// check whether tokenFile exists
+	if _, err := os.Stat(tokenFile); os.IsNotExist(err) {
+		return nil, ErrNotInCluster
+	}
+	// check whether rootCAFile exists
+	if _, err := os.Stat(rootCAFile); os.IsNotExist(err) {
+		return nil, ErrNotInCluster
```

Comment:
this changes behavior to return an error if the CA bundle is not found, instead of using the default CA bundle